src/core/server/integration_tests/saved_objects/migrations/skip_reindex.test.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import Path from 'path';
+import fs from 'fs/promises';
+import { Env } from '@kbn/config';
+import { getEnvOptions } from '@kbn/config-mocks';
+import { REPO_ROOT } from '@kbn/utils';
+import type { Root } from '@kbn/core-root-server-internal';
+import {
+  createRootWithCorePlugins,
+  createTestServers,
+  type TestElasticsearchUtils,
+} from '@kbn/core-test-helpers-kbn-server';
+import { delay } from './test_utils';
+import { SemVer } from 'semver';
+
+const logFilePath = Path.join(__dirname, 'skip_reindex.log');
+
+describe('skip reindexing', () => {
+  const currentVersion = Env.createDefault(REPO_ROOT, getEnvOptions()).packageInfo.version;
+  let esServer: TestElasticsearchUtils['es'];
+  let root: Root;
+
+  afterEach(async () => {
+    await root?.shutdown();
+    await esServer?.stop();
+    await delay(10);
+  });
+
+  it('when migrating to a new version, but mappings remain the same', async () => {
+    const { startES } = createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'basic',
+        },
+      },
+    });
+    esServer = await startES();
+    root = createRoot();
+
+    // Run initial migrations
+    await root.preboot();
+    await root.setup();
+    await root.start();
+
+    // stop Kibana and remove logs
+    await root.shutdown();
+    await delay(10);
+    await fs.unlink(logFilePath).catch(() => void 0);
+
+    const nextMinor = new SemVer(currentVersion).inc('patch').format();
```

Comment:
NIT: My bad, this should have been called `nextPatch` (I tried increasing the minor, but Kibana didn't want to start if we were one minor ahead of ES).